crates/node/api/src/node.rs

```diff
@@ -145,3 +163,19 @@ pub trait FullNodeComponents: FullNodeTypes + Clone + 'static {
     /// Returns the task executor.
     fn task_executor(&self) -> &TaskExecutor;
 }
+
+pub trait FullNodeComponentsExt: FullNodeComponents {
+    type Rpc: Rpc<Self>;
+
+    /// Returns the RPC component, if installed.
+    fn rpc(&self) -> Option<&Self::Rpc>;
+}
+
+pub trait Rpc<N: FullNodeComponents>: Send + Sync {
+    type ServerHandles;
+    type Registry;
+
+    fn handles(&self) -> &Self::ServerHandles;
+
+    fn registry(&self) -> &Self::Registry;
+}
```

Comment:
I think RPC is downstream of components and part of the `Node` / `Launch` process.

so I think we want:
* keep the rpc extension API on nodebuilder
* make rpc part of the Node helper trait, to embed it into `DefaultLauncher`

but somehow we'd need to a way to clarify the EthApi type for the nodehooks I assume?

maybe a solution that could work is is replacing the RpcHooks type with the RpcBuilder

Reply:
> we may also want to separate engine rpc and regular rpc as two components

definitely, this behaviour should be entirely node specific, hence I'm thinking this should be a separate thing and perhaps replace

https://github.com/paradigmxyz/reth/blob/38f2d00c8a25db4b46586ba00d8bbf27e6cff326/crates/node/builder/src/rpc.rs#L35-L35

and also be configured

https://github.com/paradigmxyz/reth/blob/38f2d00c8a25db4b46586ba00d8bbf27e6cff326/crates/node/builder/src/node.rs#L23-L23